pilot/pkg/config/memory/monitor.go

```diff
@@ -63,20 +66,25 @@ func NewBufferedMonitor(store model.ConfigStore, bufferSize int) Monitor {
 	}
 
 	return &configstoreMonitor{
+		running:  atomic.NewBool(false),
 		store:    store,
 		handlers: handlers,
 		eventCh:  make(chan ConfigEvent, bufferSize),
 	}
 }
 
 func (m *configstoreMonitor) ScheduleProcessEvent(configEvent ConfigEvent) {
-	m.eventCh <- configEvent
+	if m.running.Load() {
+		m.eventCh <- configEvent
+	}
 }
 
 func (m *configstoreMonitor) Run(stop <-chan struct{}) {
+	m.running.Store(true)
 	for {
 		select {
 		case <-stop:
+			m.running.Store(false)
 			if _, ok := <-m.eventCh; ok {
```

Comment:
Why do we do this? Won't this block shutdown, waiting for an event?

Reply:
Yes, it does. I fixed this and added tests